src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+                            var statements = GetStatements(currentParent);
+                            var index = statements.IndexOf(ifNode);
+
+                            var statementsBeforeIf = statements.Take(index);
+
+                            var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf).Concat(UnwrapBlock(ifNode.Statement)).Concat(newIfBody))
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    case InvertIfStyle.MoveSubsequenceToElseBody:
+                        {
+                            var currentParent = ifNode.Parent;
+                            var statements = GetStatements(currentParent);
+                            var index = statements.IndexOf(ifNode);
+
+                            var statementsBeforeIf = statements.Take(index);
+                            var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                            var updatedIf = ifNode
+                                .WithCondition(negatedCondition)
+                                .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf));
+
+                            var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf))
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    case InvertIfStyle.WithElseClause:
+                        {
+                            var currentParent = ifNode.Parent;
+                            var statements = GetStatements(currentParent);
+                            var index = statements.IndexOf(ifNode);
+
+                            var statementsBeforeIf = statements.Take(index);
+                            var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                            var updatedIf = ifNode
+                                .WithCondition(negatedCondition)
+                                .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf))
+                                .WithElse(SyntaxFactory.ElseClause(ifNode.Statement));
+
+                            var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf))
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(invertIfStyle);
+                }
+            }
+
+            private static IEnumerable<StatementSyntax> UnwrapBlock(StatementSyntax ifBody)
```

Comment:
nit: could this just return ```SyntaxList<StatementSyntax>``` instead of IEnumerable?  Feels cleaner to me.

Reply:
(resolved)